exec/cnex/nstring.c

```diff
@@ -208,6 +222,111 @@ size_t string_findChar(TString *self, char c)
     return NPOS;
 }
 
+int64_t string_findFirstOf(TString *self, size_t pos, const char *str)
+{
+    if (self->data == NULL || self->length == 0 || pos > self->length || str == NULL) {
+        return -1;
+    }
+
+    for (size_t l = pos; l < self->length; l++) {
+        if (strchr(str, self->data[l]) != NULL) {
+            return l;
+        }
+    }
+    return -1;
+}
+
+static inline BOOL isNotOneOf(TString *self, char c)
+{
+    for (size_t i = 0; i < self->length; i++) {
+        if (self->data[i] == c) {
+            return FALSE;
+        }
+    }
+    return TRUE;
+}
+
+int64_t string_findFirstNotOf(TString *self, TString *p)
```

Comment:
Why is this interface different from `string_findFirstOf`? I would have expected them to both take the same types. Should this also have a `pos` parameter?

Reply:
Modified to take a pos parameter, but left the match string as a TString object.